### ISSUE TITLE
BNode in the Ontop native mapping language 

### ISSUE THREAD

**Issue by ghxiao** [ontop/ontop#77]
Currently, one cannot specify BNode in the .obda files. This is a problem for  bootstrapped mappings generated by ontop bootstrapper.

Actually ontop is already able to handle BNode internally. Most of the RDB2RDF compliance tests with BNode in R2RML and DM scenarios passed. 

So what's missing is just a syntax for BNode in the .obda syntax and of course the parser for it.

**Comment by ghxiao** [comment]
There are two types of BNode syntax in Turtle

- (Labeled) RDF Blank Nodes: http://www.w3.org/TR/turtle/#unlabeled-bnodes:
```
_:alice foaf:knows _:bob .
```

- Unlabeled Blank Nodes http://www.w3.org/TR/turtle/#unlabeled-bnodes
```
[] foaf:knows [ foaf:name "Bob" ] .
```

Following Turtle, we will also have two syntax for BNode templates:

- Labeled RDF Blank Node Templates: 
```
_:{id}_{age}_{name}
```

- Unlabeled Blank Node Template (actually not a template since no placeholder is involved):

```
[]
```

**Comment by ghxiao** [comment]
Class needed to be changed:

- TargetQueryRenderer: for rendering Ontop Native syntax mappings
- R2RMLWriter: for rendering R2RML Mappings
- TurtleOBDA.g: for parsing Ontop Native syntax mappings

**Comment by tim-hendriks** [comment]
We are planning to use ontop within our organisation and run into this issue when attempting to bootstrap and convert to r2rml certain SQL databases (I believe tables with composite keys). Will this be included in the beta-3 release? Thanks in advance.

**Comment by bcogrel** [comment]
@ghxiao : was it already included in 3.0.1? Shall we close it?

**Comment by ghxiao** [comment]
Yes, this is already implemented.
